phoenix-core/src/it/java/org/apache/phoenix/end2end/MultiTenantTableIT.java

```diff
@@ -0,0 +1,140 @@
+        return DriverManager.getConnection(url, props);
+    }
+
+    private Connection tenantConnection(String url) throws SQLException {
+        Properties props = new Properties();
+        props.setProperty(PhoenixRuntime.CURRENT_SCN_ATTRIB, Long.toString(nextTimestamp()));
+        String tenantIdProperty = this.tenantId.replaceAll("\'", "");
+        props.setProperty(PhoenixRuntime.TENANT_ID_ATTRIB, tenantIdProperty);
+        return DriverManager.getConnection(url, props);
+    }
+
+    private final String ddl;
+    private final String dataType;
+    private final String tenantId;
+    private final String otherTenantId;
+    private final String table;
+
+    public MultiTenantTableIT( String dataType, String tenantId, String otherTenantId ) {
+        this.dataType = dataType;
+        this.tenantId = tenantId;
+        this.otherTenantId = otherTenantId;
+        String tbl = "foo" + dataType;
+        if(tbl.contains("(")){
+            tbl = tbl.substring(0, tbl.indexOf("("));
+        }
+        this.table = tbl;
+        this.ddl = "create table " + table + " (" + "tid "+ dataType + " NOT NULL," + "id INTEGER NOT NULL, \n"
+                + "val VARCHAR " + "CONSTRAINT pk PRIMARY KEY(tid, id)) \n"
+                + "MULTI_TENANT=true";
+    }
+
+    @Parameters
+    public static Collection<Object[]> data() {
+        List<Object[]> testCases = Lists.newArrayList();
+        testCases.add(new Object[] { "INTEGER", "2147483647", "2147483646" });
+        testCases.add(new Object[] { "UNSIGNED_INT", "2147483647", "2147483646" });
+        testCases.add(new Object[] { "BIGINT", "9223372036854775807", "9223372036854775806" });
+        testCases.add(new Object[] { "UNSIGNED_LONG", "9223372036854775807", "9223372036854775806" });
+        testCases.add(new Object[] { "TINYINT", "127", "126" });
+        testCases.add(new Object[] { "UNSIGNED_TINYINT", "85", "84" });
+        testCases.add(new Object[] { "SMALLINT", "32767", "32766" });
+        testCases.add(new Object[] { "UNSIGNED_SMALLINT", "32767", "32766" });
+        testCases.add(new Object[] { "FLOAT", "3.4028234", "3.4028232" });
+        testCases.add(new Object[] { "UNSIGNED_FLOAT", "3.4028234", "3.4028232" });
+        testCases.add(new Object[] { "DOUBLE", "1.7976931348623157", "1.7976931348623156" });
+        testCases.add(new Object[] { "UNSIGNED_DOUBLE", "1.7976931348623157", "1.7976931348623156" });
+        testCases.add(new Object[] { "DECIMAL", "3.402823466", "3.402823465" });
+        testCases.add(new Object[] { "VARCHAR", "\'NameOfTenant\'", "\'Nemesis\'" });
+        testCases.add(new Object[] { "CHAR(10)", "\'1234567890\'", "\'Nemesis\'" });
+
+        return testCases;
+    }
+
+    @Test
+    public void testMultiTenantTables() throws Exception {
+        Connection conn = regularConnection(getUrl());
+        conn.setAutoCommit(true);
+        conn.createStatement().execute(ddl);
+
+
+        try {
+            conn.createStatement().execute(ddl);

```

Comment:
So I think this is convention I picked up from some other tests, but the goal of that try block is to actually throw the exception and then catch it and carry on. It seemed like the way that other tests were verifying table creation was to try creating it again and verifying that the second create fails because it already exists. I do want to verify the table got created, because prior to my changes a tenantId of anything but CHAR or VARCHAR would fail at creation. If there is a better way of checking if a table exists, I would certainly be open to it though.